src/main/java/frc/robot/states/ShootState.java

```diff
@@ -0,0 +1,44 @@
+package frc.robot.states;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.util.GeometryUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.swerve.SwerveDrive;
+import java.util.Arrays;
+import lib.Utils;
+
+public class ShootState implements ScoreState {
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () -> {
+                    var alliance = DriverStation.getAlliance();
+                    var optimalPoints = Arrays.asList(Constants.OPTIMAL_POINTS_SHOOT);
+                    if (!alliance.isEmpty()) {
+                        if (alliance.get() == DriverStation.Alliance.Red) {
+                            optimalPoints.forEach(GeometryUtil::flipFieldPose);
```

Comment:
Depending on the amount of points, it might be better to save two sets of points (one for each alliance).